### PR TITLE
DO-1166 - Publish Image to dockerhub as radixdlt/babylon-node

### PR DESCRIPTION
**Pull request by kofrdx**
This PR adds to the github workflow to do the following:

- log into dockerhub and push image there if github event is "release"
- require environment: release for this
- fetch credentials from AWS

ToDo:
- [x] Merge https://github.com/radixdlt/radixdlt-iac/pull/958
- [X] Update Secret to point to AWS Main account (EDIT: Same as below. Already used the Main account on initial testing)
- [x] Rotate Dockerhub Token.  It was stored in Sandbox account and is going to be moved to Main account (EDIT: I confused Main/Sandbox account. I actually never stored the credentials in Sandbox only ever in 308190735820)

### CODE REVIEW

**Comment by kofrdx** [comment]
This PR was closed due to renaming of the branch.
https://github.com/radixdlt/babylon-node/pull/296

Relevant conversation here:
Github Actions have posted these Docker tags:
- eu.gcr.io/dev-container-repo/babylon-node:pr-296
- eu.gcr.io/dev-container-repo/babylon-node:sha-729806b

These have been created as proof that it can push towards docker-hub and can/will be deleted after approval and merge.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  cancel_running_workflows:
+    name: Cancel running workflows
+    runs-on: ubuntu-22.04
+    steps:
+      - name: cancel running workflows
+        uses: styfle/cancel-workflow-action@0.8.0
+        with:
+          access_token: ${{ github.token }}
+
+  build_deb:
```
Comment (shambupujar) [inline .github/workflows/release.yml:17]: I see we have these build steps in https://github.com/radixdlt/babylon-node/blob/feature/DO-1166-deploy-dockerhub/.github/workflows/docker.yml#L21. So you can add another job there in that gets runs for release trigger.

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -108,6 +110,25 @@ jobs:
...
+          secret-ids: |
+            DOCKERHUB, github-actions/rdxworks/dockerhub-images/release-credentials
+          parse-json-secrets: true
+        if: github.event_name == 'release'
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{env.DOCKERHUB_USERNAME}}
+          password: ${{env.DOCKERHUB_TOKEN}}
+        if: github.event_name == 'release'
```
Comment (shambupujar) [inline .github/workflows/docker.yml:131]: looks good. Although we need to define condition in multiple places, still better alternative.

**Review by shambupujar — APPROVED** [review]

**Comment by kofrdx** [comment]
Added the feedback from shambu

File: .github/workflows/release.yml (hunk 1 above)
Comment (kofrdx) [reply .github/workflows/release.yml]: updated this

**Pull request merged**